test/ast.test.js

```diff
@@ -1,74 +1,59 @@
 import { describe, expect, it } from 'vitest'
 import { tokenize, SugarHigh } from '../lib'
 
-function getTypeName(token) {
-  return SugarHigh.TokenTypes[token[0]]
-}
-
 function getTokenValues(tokens) {
   return tokens.map((tk) => tk[1])
 }
 
-function getTokenTypes(tokens) {
-  return tokens.map((tk) => getTypeName(tk))
+function getTypeName(token) {
+  return SugarHigh.TokenTypes[token[0]]
 }
 
-function extractTokensTypes(tokens) {
-  return getTokenTypes(tokens).filter(type => type !== 'space')
+function getTokenArray(tokens) {
+  return tokens.map((tk) => [tk[1], getTypeName(tk)]);
 }
 
 function mergeSpaces(str) {
   return str.trim().replace(/^[\s]{2,}$/g, ' ')
 }
 
-function filterSpaces(arr) {
-  return arr
-    .map(t => mergeSpaces(t))
-    .filter(Boolean)
-}
-
-function extractTokenValues(tokens) {
-  return filterSpaces(getTokenValues(tokens))
+function extractTokenArray(tokens) {
+  return tokens
+    .map((tk) => [mergeSpaces(tk[1]), getTypeName(tk)])
+    .filter(([_, type]) => type !== 'space' && type !== 'break')
 }
 
 describe('function calls', () => {
   it('dot catch should not be determined as keyword', () => {
     const tokens = tokenize(`promise.catch(log)`)
-    expect(extractTokensTypes(tokens)).toEqual([
-      'identifier', 'sign', 'identifier', 'sign', 'identifier', 'sign',
+    expect(getTokenArray(tokens)).toEqual([
+      ["promise", "identifier"], [".", "sign"], ["catch", "identifier"], ["(", "sign"], ["log", "identifier"], [")", "sign"]
     ])
   })
 })
 
 describe('calculation expression', () => {
   it('basic inline calculation expression', () => {
     const tokens = tokenize(`123 - /555/ + 444;`)
-    expect(getTokenTypes(tokens)).toEqual([
-      'class', 'space', 'sign', 'space', 'string', 'space', 'sign', 'space', 'class', 'sign',
-    ])
-    expect(getTokenValues(tokens)).toEqual([
-      '123', ' ', '-', ' ', '/555/', ' ', '+', ' ', '444', ';'
+    expect(getTokenArray(tokens)).toEqual([
+      ['123', 'class'], [' ', 'space'], ['-', 'sign'], [' ', 'space'], ['/555/', 'string'], [' ', 'space'],
+      ['+', 'sign'], [' ', 'space'], ['444', 'class'], [';', 'sign']
     ])
   })
 
   it('calculation with comments', () => {
     const tokens = tokenize(`/* evaluate */ (19) / 234 + 56 / 7;`)
-    expect(extractTokensTypes(tokens)).toEqual([
-      'comment', 'sign', 'class', 'sign', 'sign', 'class', 'sign', 'class', 'sign', 'class', 'sign',
-    ])
-    expect(extractTokenValues(tokens)).toEqual([
-      '/* evaluate */', '(', '19', ')', '/', '234', '+', '56', '/', '7', ';',
+    expect(extractTokenArray(tokens)).toEqual([
+      ["/* evaluate */", "comment"], ["(", "sign"], ["19", "class"], [")", "sign"], ["/", "sign"], ["234", "class"],
+      ["+", "sign"], ["56", "class"], ["/", "sign"], ["7", "class"], [";", "sign"]
     ])
   })
 
   it('calculation with defs', () => {
     const tokens = tokenize(`const _iu = (19) / 234 + 56 / 7;`)
-    expect(extractTokenValues(tokens)).toEqual([
```

Comment:
Let's keep the extractTokenValues tests as well!

Reply:
It's for visual review purpose, it's super easy to check if the tokenizer worked properly.

Reply:
I agree. With added array wrap and type it's no longer instantly clear, how string is splitted into tokens.
Our ticket evolved into: "refactor type checks to include values".
I will add commit in a minute.